File: 03-SubAgents/projects/02-test-runner/src/calculator.js
/**
 * 简单计算器模块
 * 用于演示测试运行器子代理
 */

function add(a, b) {
  return a + b;
}

function subtract(a, b) {
  return a - b;
}

function multiply(a, b) {
  return a * b;
}

function divide(a, b) {
  if (b === 0) {
    throw new Error('Division by zero');
  }
  return a / b;
}

// 故意留一个 bug：应该是 Math.sqrt
function squareRoot(n) {
  if (n < 0) {
    throw new Error('Cannot calculate square root of negative number');
  }
  return Math.sqrt(n);
}

// 故意的 bug：没有处理负数的情况
function factorial(n) {
  if (n === 0 || n === 1) return 1;
  return n * factorial(n - 1);
}

module.exports = {
  add,
  subtract,
  multiply,
  divide,
  squareRoot,
  factorial
};

File: 09-Agent-SDK/projects/04-test-fixer/src/calculator.py
"""简单计算器模块"""


def add(a, b):
    return a + b


def subtract(a, b):
    return a - b


def multiply(a, b):
    return a * b


def divide(a, b):
    if b == 0:
        raise ValueError("除数不能为零")
    return a / b


def average(numbers):
    """计算平均值 —— 注意：这里有个 bug"""
    if not numbers:
        return 0
    return sum(numbers) / (len(numbers) + 1)  # BUG: 多加了 1
